### ISSUE TITLE
Add an option to not encrypt /boot

### ISSUE THREAD

**Issue by Bleuzen** [calamares/calamares#1311]
Hi, I would really like to see an option to enable encryption in the installer, but not encrypt /boot.

This has multiple advantages:
- **decryption is much, much faster, because it can use hardware acceleration**
decryption with encrypted /boot (in grub) can take about 20 - 30 seconds on my machine and I'm not alone
- **you have multiple tries to enter your password**
this is not possible in grub currently. If you put in the wrong password, you have to reboot, which is very frustrating, because checking the password takes so long every time
- **the correct keyboard layout is loaded**
if decrypting in grub, the local key map is not loaded and so you can't use special chars in your password. Leaving /boot unencrypted and do the os do the decryption fixes this.
- **luks-2 is currently not supported in grub**
I really like to use luks-2, because, you know, it is 2020 and luks-2 is out for some time already and has some advantages I like to use. With encrypted /boot, this would currently not be possible.

Maybe there are more points, but that are the one that got me to leave /boot unencryped.

Also, all the other distros I tried so far do NOT encrypt /boot too. (examples would be Pop OS / Fedora / ...).

In my opinion, it is not worth it to encrypt boot. You are giving up too much.
If someone is really paranoid, they should store their /boot on an external usb device that they watch all day an night anyway.

So I think a checkbox in Calamares about encrypting /boot or not would be really nice.

**Comment by philmmanjaro** [comment]
Upstream is already working to add [luks-2 support](http://git.savannah.gnu.org/cgit/grub.git/log/), however I was not able to build **grub-git** yet on my end.

**Comment by Bleuzen** [comment]
@philmmanjaro Yes, it is fine that grub will finally support luks-2 soon.
But still this fixes only one of my points against encrypting /boot.
So I still would like to see an option to leave /boot unencrypted.

**Comment by kkofler** [comment]
Leaving `/boot` unencrypted leaves a huge gaping hole in that there is no authentication of the initramfs, so anybody can trojan your initramfs and steal your encrypted data as soon as you unlock it.

The initramfs may also contain sensitive data that you don't want attackers to see, e.g., decryption keys. (In particular, the current Calamares code always puts a LUKS keyfile into the initramfs as soon as you enable LUKS encryption, which is why leaving `/boot` unencrypted in manual partitioning currently defeats the whole purpose of using LUKS, though of course implementing your change request would necessarily change that.)

**Comment by apt-ghetto** [comment]
> Leaving `/boot` unencrypted leaves a huge gaping hole in that there is no authentication of the initramfs, so anybody can trojan your initramfs and steal your encrypted data as soon as you unlock it.
> 
With Secure Boot enabled (and properly handled in `shim` and GRUB), the system won't boot with a tampered `initramfs`. This is "only" a problem on BIOS installations or when Secure Boot is disabled.
> The initramfs may also contain sensitive data that you don't want attackers to see, e.g., decryption keys. (In particular, the current Calamares code always puts a LUKS keyfile into the initramfs as soon as you enable LUKS encryption, which is why leaving `/boot` unencrypted in manual partitioning currently defeats the whole purpose of using LUKS, though of course implementing your change request would necessarily change that.)

Having a keyfile in initramfs on an unencrypted _/boot_ means, that the encryption is kind of useless, because everyone can boot the system without knowing the passphrase.

In my opinion, it is worth thinking about implementing this feature request.

**Comment by Bleuzen** [comment]
@kkofler 

> Leaving `/boot` unencrypted leaves a huge gaping hole in that there is no authentication of the initramfs, so anybody can trojan your initramfs and steal your encrypted data as soon as you unlock it.

Yes, well, but isn't is also possible to inject something in grub? There has to be some software to boot which need to not be encrypted so that the system is able to boot. This could always be modified?

Also, as already said in my first post, if someone really wants to be secure, they could save their /boot and bootloader on an external usb device for example that they don't leave off of their eyes.

For a personal computer at home this is not really relevant. I mean I can see if the doors were broken when I come home. As long as everything looks untouched, I'm pretty sure that nobody got in and modified my boot partition. This should be enough for the average user at home. So it is also what the most other installers and distributions do.


> Calamares code always puts a LUKS keyfile into the initramfs as soon as you enable LUKS encryption, which is why leaving `/boot` unencrypted in manual partitioning currently defeats the whole purpose of using LUKS

Well, wait, WHAT?
Even in manual partitioning mode?
I remember someone asked me on reddit how to not encrypt boot during Manjaro setup. I answered he could use manual partitioning or the Architect installer.
I can just hope for him he didn't chose manual partitioning in Calamares.
I think, as long as this is not optional, there should be a big fat warning message if someone tries to install it this way!

**Comment by kkofler** [comment]
> Well, wait, WHAT?
> Even in manual partitioning mode?

Well, yes. Though Calamares might actually be so confused that the system will not boot at all. If it boots, it will not be secure.

See #1073.

**Comment by Bleuzen** [comment]
@apt-ghetto 
> With Secure Boot enabled (and properly handled in `shim` and GRUB), the system won't boot with a tampered `initramfs`.

I don't know much about it, but isn't it easy enough to just disable secure boot without the user noticing?
I do not check the secure boot state of my machine before every boot. Doubt many users do.

> Having a keyfile in initramfs on an unencrypted _/boot_ means, that the encryption is kind of useless, because everyone can boot the system without knowing the passphrase.

Sure. I didn't even know that Calamares is doing such things. I haven't used it for a while, but back in the days when I used Calamares to install my encrypted system, it put the keyfile directly on the root _/_ partition.
This is also the place where I put it now and should be the place where Calamares should store it if /boot is not encrypted.

> In my opinion, it is worth thinking about implementing this feature request.

:)

**Comment by kkofler** [comment]
"Secure Boot" cannot protect the initramfs. With what key would it be signed? It is generated on the user machine.

**Comment by apt-ghetto** [comment]
@Bleuzen 
> > With Secure Boot enabled (and properly handled in `shim` and GRUB), the system won't boot with a tampered `initramfs`.
> 
> I don't know much about it, but isn't it easy enough to just disable secure boot without the user noticing?

Depends on the implementation. Normally, you can set a password in the UEFI.
On some machines/firmware, you cannot disable Secure Boot (e.g. Microsoft certified ARM machines).
> I do not check the secure boot state of my machine before every boot. Doubt many users do.

Yes, but when installing, you can only check, if Secure Boot is enabled or disabled. Calamares can't know, what happens in the future.

@kkofler 
> "Secure Boot" cannot protect the initramfs. With what key would it be signed? It is generated on the user machine.

Yes, you are right. On Ubuntu, the `initrd` isn't signed and therefore it isn't validated during the boot process.
But I guess, it is technically possible to implement the validation in `shim`.

**Comment by skluthe** [comment]
It's still more secure than just a username and password. This is how a fair amount of distros do things (even Parrot which is a security focused OS) In the end, you should let your users pick how they want to use the software regardless of the security impact of this (don't make it the default so they have to know what they're doing to a point). The encrypted boot causes tons of load time issues for people and you should seriously consider adding it.

**Comment by JoelStienlet** [comment]
> "Secure Boot" cannot protect the initramfs. With what key would it be signed? It is generated on the user machine.

[This blog ](https://www.crowdstrike.com/blog/enhancing-secure-boot-chain-on-fedora-29/) explains how to protect the initramfs with secure boot on fedora, BUT it requires to replace the existing (Microsoft) keys in the UEFI, so this may not be feasible on every hardware.

**Comment by eeickmeyer** [comment]
Hi there. I'm the leader of Ubuntu Studio. Right now, we are adopting Calamares for our installer for several reasons, but this particular issue has me second-guessing.

I have received emails in the past about our Plymouth theme having difficulty with the encryption prompt box and masking response (*****), and I just switched our Plymouth theme that supports this without problem.

However, because Calamares does not support an unencrypted /boot, this feature cannot be utilized. If you look at most other installers such as Anaconda (Fedora), Ubiquity (Ubuntu), and even YaST (openSUSE), this feature of Plymouth is utilized via an unencrypted /boot.

Basically, for user-friendliness, this needs to be a feature. We hope to use Calamares as our installer in Ubuntu Studio 20.10 due out in October.

**Comment by Chrysostomus** [comment]
If you specify 
`efiSystemPartition:     "/boot"` 
in partition.conf, does that result in error or uncrypted /boot?

**Comment by kkofler** [comment]
There is no code to support this setup (full disk encryption with `efiSystemPartition: "/boot"`), so I do not expect this to work (and produce a bootable setup), and most definitely not to work securely (without decryption keys landing on the unencrypted `/boot`).

**Comment by Chrysostomus** [comment]
Okay, then it should definitely be implemented.

**Comment by Bleuzen** [comment]
Is there any progress on this?

**Comment by Chrysostomus** [comment]
I can code it if there is an indication that it could get merged.

**Comment by frapell** [comment]
I would love to see this implemented as well... I do not work for the CIA, NSA or anything like that, I just want to keep my work and private documents secure in a case someone breaks into my house and steals my desktop, or steal my backpack with my laptop... I also have Win10 on a separate partition, and I would like to be able to reach GRUB and Windows without the need to enter a password, and only have to enter it when I choose to boot Linux. This use case appears to not be possible at all with Calamares. I have tried manually partitioning, and was not able to make it work :(

**Comment by abucodonosor** [comment]
> I would love to see this implemented as well... I do not work for the CIA, NSA or anything like that, I just want to keep my work and private documents secure in a case someone breaks into my house and steals my desktop, or steal my backpack with my laptop... I also have Win10 on a separate partition, and I would like to be able to reach GRUB and Windows without the need to enter a password, and only have to enter it when I choose to boot Linux. This use case appears to not be possible at all with Calamares. I have tried manually partitioning, and was not able to make it work :(

If you NOW use unencrypted /boot with automatic unlock guess what happens if I steal your laptop?;)
( And btw even without automatic unlock )

Let me tell you, I put init=/bin/bash into grub command line, become root, change the key and steal your DATA.

If you want what you are saying you want a *full* encrypted PC.

**Comment by Chrysostomus** [comment]
>Let me tell you, I put init=/bin/bash into grub command line, become root, change the key and steal your DATA.

Root encryption even without encrypted /boot takes care of exactly this scenario. Bypassing the init system does not magically decrypt the root partition. Root powers don't enable you to decrypt anything without the encryption key. The pull request I sent prevents calamares from setting up a keyfile when /boot is unencrypted.

**Comment by apt-ghetto** [comment]
> If you NOW use unencrypted /boot with automatic unlock guess what happens if I steal your laptop?;)

What are the steps to create NOW (or in 15 minutes) an unencrypted /boot with Calamares? As far as I know, this is not possible with Calamares.

**Comment by Chrysostomus** [comment]
It is, use manual partitioning. The warning it gives does not prevent you from installing. But you don't want to do it before my pull request gets merged.

**Comment by frapell** [comment]
> If you NOW use unencrypted /boot with automatic unlock guess what happens if I steal your laptop?;)
> ( And btw even without automatic unlock )
> 
> Let me tell you, I put init=/bin/bash into grub command line, become root, change the key and steal your DATA.
> 
> If you want what you are saying you want a _full_ encrypted PC.

What? What would be the point on having `/boot` unencrypted and automatic unlock enabled? I wouldn't be using encryption in that case...

I would LOVE to have something that is perfectly doable, and I have done every time I configure a new machine. I encrypt an LVM partition and do not setup any form of automatic unlocking. Inside the LVM I can do whatever I want partition-wise, but I usually have a root partition, a home partition and a work partition. Then, when I turn on my machine, I get GRUB I can choose Windows and boot, or choose Linux AND THEN I am asked for a password (with even a nice UI) to decrypt my LVM and continue booting. My understanding is that this is perfectly safe, and if someone steals my machine, yeah, they will be able to get to Windows, but not Linux. Please correct me if I am wrong...

**Comment by Chrysostomus** [comment]
It is not "perfectly safe", but sufficiently safe for most purposes. The attack vector that encrypted /boot protects you from is the evil maid attack. But if someone just steals your computer, your data is still safe.

**Comment by frapell** [comment]
@Chrysostomus Exactly... Plus, never forget to be pragmatic

https://xkcd.com/538/

**Comment by mariomie** [comment]
How do I need to setup the partitions to end up with /boot not encrypted?
Thanks,

**Comment by Bleuzen** [comment]
@mariomie You have to use manual partitioning. Just create one partition mounted at /boot but without ticking the encrypt checkbox like this:
https://www.youtube.com/watch?v=ayM3PTnuOCQ

**Comment by blackandyellowgecko** [comment]
Great that this is being addressed! GRUB decryption screen has a lot of UX problems:

1. If you press enter (accidentally or not) while typing the password and start typing again, GRUB will "leak" your password in the screen while entering GRUB rescue. While decryption is in progress, keyboard should not register anything aside from keyboard shortcuts.

2. Why just one incorrect attempt makes the GRUB enter in rescue mode? Three attempts is a standard and it's better. Also, GRUB could show a keyboard shortcut like "press R to enter rescue mode".

3. There is no message indicating that decrypting is in progress. You if don't press enter correctly, you won't know if the decryption started or not.

4. Password characters doesn't appear in the screen as asterisks while you typing. This is very useful to check if the system registered two characters instead of one.

5. Backspace doesn't work.

**Comment by blackandyellowgecko** [comment]
> @mariomie You have to use manual partitioning. Just create one partition mounted at /boot but without ticking the encrypt checkbox like this:
> https://www.youtube.com/watch?v=ayM3PTnuOCQ

I tried doing this in a VM, but the bootscreen keeps showing the loading spin and the password box never appears.

![Uploading Screenshot_ubuntu20.10_2021-05-07_05:22:39.png…]()
![Screenshot_ubuntu20 10_2021-05-07_05:29:36](https://user-images.githubusercontent.com/78433932/117421854-528b1800-aef5-11eb-974e-75bc94193373.png | width =500)

**Comment by lyrebird-sound-cloning-system** [comment]
@

> @mariomie You have to use manual partitioning. Just create one partition mounted at /boot but without ticking the encrypt checkbox like this:
> https://www.youtube.com/watch?v=ayM3PTnuOCQ

This doesn't work, unfortunately. The user ends up in a busy box.

**Comment by Bleuzen** [comment]
> Ubuntu and Manjaro creates LVM when setting up encrypted disk, shouldn't this be done too?

No, LVM is optional.

> > https://www.youtube.com/watch?v=ayM3PTnuOCQ
> 
> This doesn't work, unfortunately. The user ends up in a busy box.

Weird. Last time I tried this on EndeavourOS it worked fine. Maybe this does not work on other distros (with slightly older versions of Calamares?) yet.

**Comment by lyrebird-sound-cloning-system** [comment]
> Weird. Last time I tried this on EndeavourOS it worked fine. Maybe this does not work on other distros (with slightly older versions of Calamares?) yet.

Neon 5.21.4 is using version 3.2.39.3.

**Comment by lyrebird-sound-cloning-system** [comment]
I think that the this option should appear in this screen:

https://github.com/calamares/calamares/issues/1699

There would be more space to explain why turn this on or off.

**Comment by lyrebird-sound-cloning-system** [comment]
I think that the this option should appear in this screen: #1699 There would be more space to explain why turn this on or off.

Edit: wouldn't this issue be resolved if there was an option to use systemd-bootloader? It decrypts the disk much faster.

**Comment by sezanzeb** [comment]
It has been close to two years now, and the issue existed even before that. Calamares doesn't even offer the option for /boot anymore and I can't boot the resulting installation if I try to leave /boot unencrypted.

I don't want to go through the pain each time if I have a typo in my password of either rebooting my computer or trying to type in the command to retry it, with the wrong keyboard layout in place.

please implement this

**Comment by dalto8** [comment]
> It has been close to two years now, and the issue existed even before that. Calamares doesn't even offer the option for /boot anymore.
You can use manual partitioning.  If your distro doesn't have `/boot` in the dropdown, you can type `/boot` in that box.


 and I can't boot the resulting installation if I try to leave /boot unencrypted.



> I can't boot the resulting installation if I try to leave /boot unencrypted.

This works in the distros I have tried recently.  I have done quite a few installs this way.  Which distro is it failing for you on?

**Comment by sezanzeb** [comment]
Manjaro. Do I need to specify /boot/efi separately?

**Comment by dalto8** [comment]
> Do I need to specify /boot/efi separately?

These are separate things.
* Having an unenecrpyted `/boot` using a filesystem such as ext4 to allow your initrd to decrypt the luks root instead of grub
* The location of the efi partition

The distro decides the location of the efi partition.  For distros which use grub, this is usually `/boot/efi`.

So, if you partition manually and want an unencrypted `/boot` you should create a `/boot` using ext4 or similar and create a `/boot/efi` partition formatted with fat32/vfat and marked as boot/efi.

Of course, that assumes you are doing a UEFI install.  If you are doing a legacy install, you don't need an EFI partition.

**Comment by sezanzeb** [comment]
Thanks. Which partition gets the boot flag?

**Comment by dalto8** [comment]
> Thanks. Which partition gets the "boot" flag?

In a UEFI install, `/boot/efi` is the partition that should have that flag set.

**Comment by kkofler** [comment]
(And in a BIOS on GPT install, it is the BIOS Boot Partition. The `/boot` partition should never get the boot flag, it contains the files that are loaded in the next step of booting, after booting from the actual boot partition (EFI System Partition or BIOS Boot Partition).)

**Comment by sezanzeb** [comment]
getting a new usb stick with a newer version fixes the issue. For future reference, here is the setup:

- `/boot/efi`, 550MB, fat32, "boot" flat
- `/boot/`, ext4, 1000MB
- `/`, ext4, encrypted

Thanks for your quick response. Sorry for not trying out new versions first. Thanks for the clarifications

**Comment by paultstone** [comment]
I tried to do this manually on Manjaro (unencrypted /boot partition), and I got this error during the installation process:

Installation Failed
External command finished with exit code 1.
Output:
==> Building image from preset: /etc/mkinitcpio.d/linux513.preset:'default'
-> -k /boot/vmlinuz-5.13-x86_64 -c /etc/mkinitcpio.conf -g /boot/initramfs-5.13-x86_64.img
==> Starting build: 5.13.19-2 MANJARO
-> Running build hook: [base]
==> Creating gzip-compressed initcpio image: /boot/initramfs-5.13-x86_64-fallback.img
==> WARNING: errors were encountered during the build. The image may not be complete.

And the resulting system indeed does not boot.

**Comment by dalto8** [comment]
> /boot/uefi    400 MB FAT32   (boot flag)
> /boot            1200 MB ext4
> /root             XXXX MB LUKS encrypted ext4
> (n/a)             70000 MB LUKS encrypted ext4      (swap flag)

It needs to be `/boot/efi` not `/boot/uefi` and `/` instead of `/root`

**Comment by paultstone** [comment]
Thanks. :-)

`/root` was a typo. I meant `/`

I will double-check on the other one.

**Comment by paultstone** [comment]
Seems like I had to specify the boot partition as /boot/, with the trailing slash.

That works, but now I’m prompted to unlock the root partition twice. The first time doesn’t matter. I can enter anything, but then it delays for a long time before actually starting to boot. Here is the first prompt I get, which may be a BIOS thing?

Attempting to decrypt master key.
Enter passphrase for hd0,8pt3 (5662a3146e04549a1a5722135ccb117) :

**Comment by sezanzeb** [comment]
Try to remove `GRUB_ENABLE_CRYPTODISK=y` from /etc/default/grub

**Comment by paultstone** [comment]
That works! Thanks!

There was an extra GRUB_ENABLE_CRYPTODISK=Y which was added at the end of /etc/default/grub (in addition to an identical line which was commented out above in the file). As root, I edited the file and prepended `#` in front of that line at the end of the file. Then, I did the following:

sudo update-grub

I actually got an error that it didn't recognize partition nvme01, but otherwise the command seemed to complete OK, and when I rebooted it came up fine and without redundantly asking me for the passphrase before loading the bootloader. Excellent!

I also had enabled auto-login, and that works as well. I'm only prompted once for the passphrase to unlock the `/` (root) partition.

**Comment by wnhre2ur8cxx8** [comment]
> That works! Thanks!
> 
> There was an extra GRUB_ENABLE_CRYPTODISK=Y which was added at the end of /etc/default/grub (in addition to an identical line which was commented out above in the file). As root, I edited the file and prepended `#` in front of that line at the end of the file. Then, I did the following:
> 
> sudo update-grub
> 
> I actually got an error that it didn't recognize partition nvme01, but otherwise the command seemed to complete OK, and when I rebooted it came up fine and without redundantly asking me for the passphrase before loading the bootloader. Excellent!
> 
> I also had enabled auto-login, and that works as well. I'm only prompted once for the passphrase to unlock the root partition.

Beware, @paultstone  I just tried that setup and there is a major security issue with it.

Absurdly enough the installation works when entering a trailing slash to the boot mount point. But the system still thinks that your /boot is encrypted. So it stores the keyfile `/crypto_keyfile.bin` directly into your initramfs image. Which is NOT encrypted anymore. That means you have an encrypted partition with the key for it right beneath it on an completely unencrypted partition. I realized that when I was not prompted for my encryption password anymore. Only the user password. I think you mixed the prompts up.

**Comment by wnhre2ur8cxx8** [comment]
Basically all the "tutorials" on how to not encrypt your /boot partition with calamares lead to the stated outcome I fear. They are very unsafe, and should not be used!

The devs here should implement a solution to leave /boot unencrypted instead. Its needed for many setups where grub decryption is simply is too inflexible. For example decrypting your system with yubikey or similar tools.

**Comment by kkofler** [comment]
The issue where Calamares would write a keyfile on an unencrypted `/boot` is supposed to have been fixed for a long time. Why is it apparently still happening?

**Comment by wnhre2ur8cxx8** [comment]
Usually the installation with an unencrypted /boot just doesnt work. It fails at the mkinipcio step. Apparently you can trick it by entering a trailing / to the mountpoint.

**Comment by dalto8** [comment]
> Usually the installation with an unencrypted /boot just doesnt work. It fails at the mkinipcio step.

What ISO are you using?  I have done many installs with an unencrypted `/boot`

**Comment by wnhre2ur8cxx8** [comment]
latest manjaro gnome

**Comment by dalto8** [comment]
> Apparently you can trick it by entering a trailing / to the mountpoint.

If you add a trailing `/` to `/boot` manually, that will cause the detection of an unencrypted `/boot` to fail.  That is a bug that probably should be fixed.

If nobody gets to it first, I will submit a PR for that in the next couple of days.

**Comment by wnhre2ur8cxx8** [comment]
I just removed the keyfile slot from the luks partition and added my own, removed the keyfile from the initramfs and also removed it from crypttab, now its properly encrypted again and asks me for the phrase after grub. So I am sure that the setup is still as before.

Can you at least exchange the warning saying "its insecure not encrypting /boot etc" with something that says its not possible at the moment and you have to do it manually or something? So people dont have to guess around and make stupid mistakes? :)

**Comment by dalto8** [comment]
> Can you at least exchange the warning saying "its insecure not encrypting /boot etc" with something that says its not possible at the moment and you have to do it manually or something? So people dont have to guess around and make stupid mistakes? :)

It is only broken because you are manually adding the `/` at the end.  It is just as easy to fix that as add the warning.

**Comment by dalto8** [comment]
> latest manjaro gnome

I just tested this and I can't replicate your issue.  There is no need to add that trailing slash.  I tried two different Calamares ISOs one of which was the latest Manjaro Gnome and both worked without any mkinitcpio errors.

![image](https://user-images.githubusercontent.com/57767042/154168091-86929a7e-0e80-4223-a274-91c149cdcd33.png)

**Comment by wnhre2ur8cxx8** [comment]
Strange. Well I had the exact same issue as described here: https://github.com/calamares/calamares/issues/1311#issuecomment-988380404 

And also every time I tried the install in all the different ways. Just to get /boot not encrypted. Which is imho a very good reason to add some warning that the software is simply not working the way you want it to. But of course thats not my decision. The iso I downloaded is 2 days old, so dont know what else to say :)

**Comment by mariomie** [comment]
It is not working for me with the manjaro GNOME version, too. However the manjaro KDE version is working. I do not know why and posted this in the manjaro forum [https://forum.manjaro.org/t/install-without-boot-encryption-working-with-kde-version-but-not-with-gnome-version/102089](url)

**Comment by dalto8** [comment]
There are multiple issues being discussed here:
* Calamares supporting an unencrypted `/boot`. Currently Calamares supports an unencrypted `/boot` when you use manual partitioning
* The keyfile is placed in the initramfs when you use `/boot/` instead of `/boot`.  This is a bug in my personal opinion.  I am testing a fix right now.  I will do a PR for it later today unless it is fixed by someone else before I have a chance.
* Some Manjaro installs are failing on mkinitcpio when an unecrypted `/boot` is selected.  I would speculate that is not a Calamares issue, but a distro-specific issue based on the feedback above and the number of installs I have personally done with an unencrypted `/boot`.  That being said, I couldn't replicate the issue with the Manjaro gnome ISO so I can't say for sure.

**Comment by paultstone** [comment]
I am using Manjaro GNOME, and I can confirm this is true for me as well:

> Beware, @paultstone I just tried that setup and there is a major security issue with it.
> 
> Absurdly enough the installation works when entering a trailing slash to the boot mount point. But the system still thinks that your /boot is encrypted. So it stores the keyfile /crypto_keyfile.bin directly into your initramfs image. Which is NOT encrypted anymore. That means you have an encrypted partition with the key for it right beneath it on an completely unencrypted partition. I realized that when I was not prompted for my encryption password anymore. Only the user password. I think you mixed the prompts up.

**Comment by wnhre2ur8cxx8** [comment]
> There are multiple issues being discussed here:
> 
>     * Calamares supporting an unencrypted `/boot`. Currently Calamares supports an unencrypted `/boot` when you use manual partitioning
> 
>     * The keyfile is placed in the initramfs when you use `/boot/` instead of `/boot`.  This is a bug in my personal opinion.  I am testing a fix right now.  I will do a PR for it later today unless it is fixed by someone else before I have a chance.
> 
>     * Some Manjaro installs are failing on mkinitcpio when an unecrypted `/boot` is selected.  I would speculate that is not a Calamares issue, but a distro-specific issue based on the feedback above and the number of installs I have personally done with an unencrypted `/boot`.  That being said, I couldn't replicate the issue with the Manjaro gnome ISO so I can't say for sure.

Thanks for the clear summary. Now the whole thing makes sense. I will report this to the manjaro devs aswell. I thought that calamares is also a manjaro product.

But one thing is still unclear to me: Why is this issue still open when the option is already implemented? :smile:

**Comment by dalto8** [comment]
I submitted PR https://github.com/calamares/calamares/pull/1888 for the `/boot/` issue.



> Why is this issue still open when the option is already implemented? 

I don't the answer to this but my guess would be that it is because auto partitioning doesn't support it, only manual.

**Comment by abalfoort** [comment]
To tackle the unencrypted /boot partition issue I created a module for our calamares-settings-solydxk package. It also configures a previously encrypted partition that is mounted as-is (like /home). In that case it will ask for an existing paraphrase of that partition. It would be better if this would be part of Calamares but now the user is not left with an unbootable system.

If anyone's interested, you can find it here: https://gitlab.com/abalfoort/calamares-settings-solydxk/-/tree/master/calamares-modules/luksboot-fix

**Comment by demmm** [comment]
>To tackle the unencrypted /boot partition issue I created a module for our calamares-settings-solydxk package

Pull Requests are always appreciated and encouraged.

**Comment by adriaandegroot** [comment]
Presumed closed with @abalfoort 's new code (thanks! I'm trusting you to test it with some current-ish distro)

**Comment by sezanzeb** [comment]
> Hi, I would really like to see an option to enable encryption in the installer, but not encrypt /boot.

As far as I can tell in the pull request https://github.com/calamares/calamares/pull/1931 no GUI elements were added. Are you sure this issue should be closed? Or has this option been added as well in a different PR or something?

**Comment by dalto8** [comment]
> > Hi, I would really like to see an option to enable encryption in the installer, but not encrypt /boot.
> 
> As far as I can tell in the pull request #1931 no GUI elements were added. Are you sure this issue should be closed? Or has this option been added as well in a different PR or something?

That has been possible for some time.  Just use manual partitioning and don't encrypt `/boot`

**Comment by sezanzeb** [comment]
I think the point was to have something user friendly that doesn't require digging through online discussions to figure out sizes, file system, mountpoints and flags.

Something like a checkbox to tick.

Would you like to have a new issue to keep track of that feature request? Then I'll create one

**Comment by wnhre2ur8cxx8** [comment]
Since it is possible to not encrypt /boot I would create a new issue regarding making the option more accessible and maybe link to this issue here.

**Comment by adrelanos** [comment]
Could you explain please how to use the "not encrypt /boot" feature please? Created

* https://github.com/calamares/calamares/issues/2308

for it.
